scipy/special/tests/test_trig.py

```diff
@@ -57,8 +56,123 @@ def test_intermediate_overlow():
         assert_allclose(res.real, std.real)
         assert_allclose(res.imag, std.imag)
 
+def test_sinpi():
```

Comment:
I didn't look very closely, but are there test cases for input `nan`, `+inf`, and `-inf`? They could be useful for completeness.